src/domain/relay/relayers/no-fee-campaign.relayer.ts

```diff
@@ -0,0 +1,255 @@
+import { Inject, Injectable } from '@nestjs/common';
+import type { Address } from 'viem';
+import { IRelayer } from '@/domain/relay/interfaces/relayer.interface';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { IRelayApi } from '@/domain/interfaces/relay-api.interface';
+import { LimitAddressesMapper } from '@/domain/relay/limit-addresses.mapper';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { Relay, RelaySchema } from '@/domain/relay/entities/relay.entity';
+import { RelayLimitReachedError } from '@/domain/relay/errors/relay-limit-reached.error';
+import { BalancesService } from '@/routes/balances/balances.service';
+import {
+  NoFeeCampaignConfiguration,
+  RelayRules,
+} from '@/domain/relay/entities/relay.configuration';
+
+@Injectable()
+export class NoFeeCampaignRelayer implements IRelayer {
+  private readonly relayconfiguration: NoFeeCampaignConfiguration;
+  private static readonly DEFAULT_FIAT_CODE = 'USD';
+
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IConfigurationService) configurationService: IConfigurationService,
+    private readonly limitAddressesMapper: LimitAddressesMapper,
+    @Inject(IRelayApi) private readonly relayApi: IRelayApi,
+    @Inject(BalancesService) private readonly balancesService: BalancesService,
+  ) {
+    this.relayconfiguration = configurationService.getOrThrow(
+      'relay.noFeeCampaign',
+    );
+  }
+
+  async canRelay(args: {
+    chainId: string;
+    address: Address;
+  }): Promise<{ result: boolean; currentCount: number; limit: number }> {
+    const noFeeCampaignConfigurationPerChain =
+      this.relayconfiguration[parseInt(args.chainId)];
```

Comment:
```suggestion
      this.relayConfiguration[parseInt(args.chainId)];
```

It should also be refactored in other places.